management/src/main/java/org/ehcache/management/registry/DefaultCapabilityBasedQuery.java

```diff
@@ -0,0 +1,302 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.management.registry;
+
+import org.ehcache.management.CallQuery;
+import org.ehcache.management.CapabilityBasedQuery;
+import org.ehcache.management.CapabilityManagement;
+import org.ehcache.management.ContextualResult;
+import org.ehcache.management.ContextualStatistics;
+import org.ehcache.management.ResultSet;
+import org.ehcache.management.StatisticQuery;
+import org.ehcache.management.providers.ManagementProvider;
+import org.terracotta.management.stats.Statistic;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Mathieu Carbou
+ */
+final class DefaultCapabilityBasedQuery implements CapabilityBasedQuery {
+
+  private final String capabilityName;
+  private final CapabilityManagement capabilityManagement;
+
+  DefaultCapabilityBasedQuery(CapabilityManagement capabilityManagement, String capabilityName) {
+    this.capabilityManagement = capabilityManagement;
+    this.capabilityName = capabilityName;
+  }
+
+  @Override
+  public StatisticQuery.Builder queryStatistic(String statisticName) {
+    return new StatisticQueryBuilder(capabilityManagement, capabilityName, Collections.singletonList(statisticName));
+  }
+
+  @Override
+  public StatisticQuery.Builder queryStatistics(String... statisticNames) {
+    return new StatisticQueryBuilder(capabilityManagement, capabilityName, Arrays.asList(statisticNames));
+  }
+
+  @Override
+  public StatisticQuery.Builder queryStatistics(Collection<String> statisticNames) {
+    return new StatisticQueryBuilder(capabilityManagement, capabilityName, statisticNames);
+  }
+
+  @Override
+  public CallQuery.Builder call(String methodName, String[] argClassNames, Object[] args) {
+    return new CallQueryBuilder(capabilityManagement, capabilityName, methodName, argClassNames, args);
+  }
+
+  @Override
+  public CallQuery.Builder call(String methodName) {
+    return call(methodName, new String[0], new Object[0]);
+  }
+
+  static class StatisticQueryBuilder implements StatisticQuery, StatisticQuery.Builder {
+
+    private final CapabilityManagement capabilityManagement;
+    private final String capabilityName;
+    private final Collection<String> statisticNames;
+    private final List<Map<String, String>> contexts;
+    private final long since;
+
+    StatisticQueryBuilder(CapabilityManagement capabilityManagement, String capabilityName, Collection<String> statisticNames) {
+      this(capabilityManagement, capabilityName, statisticNames, Collections.<Map<String, String>>emptyList(), 0);
```

Comment:
Since it is a timestamp, we can set System.currentTimeMillis() ? It would work because at the time of creation, the stat framework is not yet started so we won't miss some stats.